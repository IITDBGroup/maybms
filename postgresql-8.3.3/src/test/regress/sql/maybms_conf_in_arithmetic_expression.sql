--test for multiple conf in a query

create table r (type varchar, p int);
insert into r values('a', 1), ('a', 2), ('b', 1), ('b', 2);

create table C as repair key type in r weight by p; 

--This tree queries should work after Lyublena disable the check for it
select conf() + 1 as p from C;

select conf()/8 as p from C;

select (conf()+1)/7 as p from C;
--

select cast( (conf()+1)/7 as real ) as p from C;

select cast( 9+(conf()+1)/7 as real ) as p from C;

select cast( abs(0-(conf()+1)/7) as real ) as p from C;

drop table r;
drop table C;
