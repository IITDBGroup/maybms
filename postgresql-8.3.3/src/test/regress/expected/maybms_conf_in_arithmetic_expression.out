--test for multiple conf in a query
create table r (type varchar, p int);
insert into r values('a', 1), ('a', 2), ('b', 1), ('b', 2);
create table C as repair key type in r weight by p; 
--This tree queries should work after Lyublena disable the check for it
select conf() + 1 as p from C;
 p 
---
 2
(1 row)

select conf()/8 as p from C;
   p   
-------
 0.125
(1 row)

select (conf()+1)/7 as p from C;
         p         
-------------------
 0.285714285714286
(1 row)

--
select cast( (conf()+1)/7 as real ) as p from C;
    p     
----------
 0.285714
(1 row)

select cast( 9+(conf()+1)/7 as real ) as p from C;
    p    
---------
 9.28571
(1 row)

select cast( abs(0-(conf()+1)/7) as real ) as p from C;
    p     
----------
 0.285714
(1 row)

drop table r;
drop table C;
